polygeist/tools/cgeist/driver.cc

```diff
@@ -900,64 +931,70 @@ static LogicalResult compileModule(mlir::OwningOpRef<mlir::ModuleOp> &Module,
     if (!LLVMModule) {
       Module->dump();
       llvm::errs() << "Failed to emit LLVM IR\n";
+      cleanUpOpenFile();
       return failure();
     }
 
     LLVMModule->setDataLayout(DL);
     LLVMModule->setTargetTriple(Triple.getTriple());
+
     LLVM_DEBUG(llvm::dbgs()
                << "*** Translated MLIR to LLVM IR successfully ***\n");
 
-    if (EmitBC || EmitLLVM) {
-      // Not needed when emitting binary for now; will be handled by the driver.
-      runOptimizationPipeline(*LLVMModule, options);
```

Comment:
After this change:
```
/iusers/waihungt/llvm/polygeist/tools/cgeist/driver.cc:886:13: warning: ‘void runOptimizationPipeline(llvm::Module&, Options&)’ defined but not used [-Wunused-function]
  886 | static void runOptimizationPipeline(llvm::Module &Module, Options &options) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~
```
Is `runOptimizationPipeline` no longer needed?

Reply:
yes... I'll make a PR to remove that

Reply:
https://github.com/intel/llvm/pull/10686